backend/tests/v5/unit/processors/teamspaces/projects/containers/containers.test.js

```diff
@@ -118,6 +134,62 @@ const testGetContainerList = () => {
 	});
 };
 
+const testAppendFavourites = () => {
+	describe('Add containers to favourites', () => {
+		test('user favourites should stay the same if one or more containers is not found', async () => {
+			await Containers.appendFavourites('tsAdmin', 'teamspace', 'project', [1, -1]);
+			expect(user1Favourites).toEqual([1]);
```

Comment:
again you're not really testing anything within the codebase, you're essentially testing the mockImplementation you defined in this file.

test these round trip logic in the e2e tests, where you're dealing with real data. The whole point of unit tests are to test that the precondition and postcondition matches.

so in this case, appendFavourites should just work if you put in valid data, so the test you should be testing is:

1. favourites should update fine if the user has access to all the models
2. should throw an error if any if the models you cannot add to to the favourites
3. deal with edge cases - what if the list is empty?
4. what if the functions this function call throws an error -> do we deal with them appropriately?